tests/lib/cli-engine.js

```diff
@@ -927,6 +927,141 @@ describe("CLIEngine", function() {
             });
         });
 
+        describe("useCache", function() {
+            var sandbox;
+            var fs = require("fs");
+
+            function delCache() {
+                try {
+                    fs.unlinkSync(path.resolve(".eslintcache"));
+                } catch(ex) {
+                    // we don't care if the file didn't exist
+                    // since our intention was to remove the file
+                }
+            }
+
+            beforeEach(function() {
+                delCache();
+                sandbox = sinon.sandbox.create();
+            });
+            afterEach(function() {
+                sandbox.restore();
+                delCache();
+            });
+            it("should remember the files from a previous run and do not operate on them if not changed", function() {
+
+                assert.isFalse(fs.existsSync(path.resolve(".eslintcache")), "the cache for eslint does not exist");
+
+                engine = new CLIEngine({
+                    useEslintrc: false,
+                    // specifying useCache true the cache will be created
+                    useCache: true,
+                    rules: {
+                        "no-console": 0,
+                        "no-unused-vars": 2
+                    },
+                    extensions: ["js"]
+                });
+
+                var spy = sandbox.spy(fs, "readFileSync");
+
+                var file = getFixturePath("cache/src", "test-file.js");
+
+                engine.executeOnFiles([file]);
```

Comment:
Yes. `executeFiles()` should return the same results regardless of cache being on or off.